infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -160,7 +160,15 @@
     public ServerContext(Map<String, Object> serverConfig) {
         this.serverConfig = serverConfig;
         this.dataStore = new DataStore(serverConfig, this::dataStoreFileCleanup);
-        generateNodeId();
+
+        if (serverConfig.get("--node-id") != null) {
```

Comment:
Curious, why is this needed?

Reply:
@annym The current lock holder replication server is identified by its clientId/nodeId. Because I need a way to kill the replication server with a specific clientId -- the one that is holding the lock at the moment, I need to know this information beforehand. Otherwise, the clientId is assigned randomly and I am not sure how to extract this information once the server is running.

Reply:
Ok, sounds reasonable. Would you mind adding something in this line as a comment in the code for future reference? thanks!